api/src/test/java/org/openmrs/ObsTest.java

```diff
@@ -331,6 +331,44 @@ public void setValueAsString_shouldFailIfTheValueOfTheStringIsNull() throws Exce
 		Obs obs = new Obs();
 		obs.setValueAsString(null);
 	}
+
+	/**
+	 * @see Obs#setValueAsString(String)
+	 */
+	@Test
+	public void setValueAsString_shouldSetValueAsDateTimeIfDataTypeOfQuestionConceptIsDateTimeHavingTimeZone()
```

Comment:
Where do you check if the obs set the date correctly? Also this may be useful for exception testing:https://github.com/junit-team/junit4/wiki/exception-testing

Reply:
I have edited the code such that it checks if the obs have been set properly by checking for the presence of the exception that would be thrown if they are improperly set